base/server/python/pki/server/cli/plugin.py

```diff
@@ -0,0 +1,762 @@
+            elif o == '--removePin':
+                remove_pin = True
+            elif o == '--attr':
+                pin_attr = a
+            elif o == '--ldapAttrName':
+                attr_name = a
+            elif o == '--ldapAttrDesc':
+                attr_desc = a
+            else:
+                print('ERROR: unknown option ' + o)
+                self.print_help()
+                sys.exit(1)
+
+        instance = pki.server.PKIInstance(instance_name)
+        if not instance.is_valid():
+            print('ERROR: Invalid instance %s.' % instance_name)
+            sys.exit(1)
+
+        instance.load()
+
+        subsystem_name = self.parent.parent.parent.name
+        subsystem = instance.get_subsystem(subsystem_name)
+
+        if not subsystem:
+            print('ERROR: No %s subsystem in instance %s.'
+                  % (subsystem_name.upper(), instance_name))
+            sys.exit(1)
+
+        default_plugin_list = [i.split(".")[2].lower() for i, j in subsystem.config.items()
+                               if i.startswith('auths.impl')]
+
+        plugin_base = ''
+
+        if not plugin_name:
+            print("ERROR: Missing plugin name.")
+            self.print_help()
+            sys.exit(1)
+        else:
+            plugin_base = 'auths.instance.{}.'.format(plugin_name)
+
+        if not plugin_type:
+            print("ERROR: Missing plugin type.")
+            self.print_help()
+            sys.exit(1)
+        else:
+            if plugin_type.lower() not in default_plugin_list:
+                print("ERROR: Not valid plugin type")
+                sys.exit(1)
+            configure_plugin[plugin_base + 'pluginName'] = plugin_type
+
+        if plugin_type.lower() in ['uidpwddirauth', 'uidpwdpindirauth', 'sharedtoken']:
+
+            if not host:
+                print("ERROR: Missing ldap hostname.")
+                self.print_help()
+                sys.exit(1)
+            else:
+                configure_plugin[plugin_base + 'ldap.ldapconn.host'] = host
+
+            if not port:
+                print("ERROR: Missing ldap port.")
+                self.print_help()
+                sys.exit(1)
+            else:
+                configure_plugin[plugin_base + 'ldap.ldapconn.port'] = port
+
+            if not dn_pattern:
+                print("ERROR: Missing ldap DN Pattern.")
+                self.print_help()
+                sys.exit(1)
+            else:
+                configure_plugin[plugin_base + 'dnpattern'] = dn_pattern
+
+            if not base_dn:
+                print("ERROR: Missing ldap basedn")
+                self.print_help()
+                sys.exit(1)
+            else:
+                configure_plugin[plugin_base + 'ldap.basedn'] = base_dn
+
+            if not secure_conn:
+                configure_plugin[plugin_base + 'ldap.ldapconn.secureConn'] = 'false'
+            else:
+                configure_plugin[plugin_base + 'ldap.ldapconn.secureConn'] = 'true'
```

Comment:
I think the default value for `secure_conn` should be `false`, which can be changed with a CLI option, so here we just simply store the value, no need for an `if-then` statement:
```
configure_plugin[plugin_base + 'ldap.ldapconn.secureConn'] = secure_conn
```
Same thing for the `if-then` statements below.